torch/optim/sparse_adam.py

```diff
@@ -97,10 +97,47 @@ def step(self, closure=None):
 
         return loss
 
-SparseAdam.__doc__ = r"""Implements lazy version of Adam algorithm suitable for sparse tensors.
+SparseAdam.__doc__ = r"""SparseAdam implements a masked version of the Adam algorithm
+    suitable for sparse gradients. This is a semantically different optimizer that
+    approximates the Adam algorithm by masking out the parameter and moment updates
+    corresponding to the zero values in the gradients. Whereas the Adam algorithm will
+    update the first moment, the second moment, and the parameter based on all values
+    of the gradients, SparseAdam only updates the moments and parameters corresponding
+    to the non-zero values of the gradients.
+    
+    A simplified way of thinking about the implementation is as such:
+    1. Create a mask of the non-zero values in the sparse gradients. For example,
+    if your gradient is [0, 5, 0, 0, 9], the mask would be [0, 1, 0, 0, 1].
+    2. Apply this mask over the running moments and do computation on only the
+    non-zero values.
+    3. Apply this mask over the parameters and only apply an update on non-zero values.
+    
+    We use sparse layout Tensors to optimize this approximation, which means the more
+    gradients that are masked by being set to 0s, the more performant the optimization.
```

Comment:
nit: mention here that part of this optimization is that we detect what is 0 based on the non-materialized value in the sparse Tensor, we don't actually check the value.